### PR TITLE
Feature/generate qos string 63

### LINKED ISSUE #63: Add logic in the build scripts to generate the qos_string.h files every time we compile
**TODO**:
- [x] Support for build.sh script
- [x] Support for build.bat script
- [x] First CR [Code Review](https://github.com/rticommunity/rtiperftest/pull/88)
- [x] Documentation
- [ ] Full [Code Review](https://github.com/rticommunity/rtiperftest/pull/88)
- [ ] Final Merge into master

### PR DESCRIPTION
**Pull request by softwarejimenez**
This pull request is coming from the issue #63. When the pull request is merged into "master" the issue will be resolved #63

### CODE REVIEW

**Comment by softwarejimenez** [comment]
@javmorcas @alfonso-soto can you take a look at it?

File: build.bat [hunk 1]
```diff
@@ -10,6 +10,8 @@ set "cs_folder=%script_location%srcCs"
 set "java_folder=%script_location%srcJava"
 set "java_scripts_folder=%script_location%resource\java_scripts"
 set "bin_folder=%script_location%bin"
+set "cStringifFile_script=%script_location%resource\perl_script\cStringifyFile.pl"
```
Comment (javmorcas) [inline build.bat]: cStringifyFile

File: build.bat [hunk 2]
```diff
@@ -153,6 +156,28 @@ if !BUILD_JAVA! == 1 (
 )
 ::------------------------------------------------------------------------------
 
+if !BUILD_CPP! == 1 (
+	call !PERL! -version > nul 2>nul
+	if not !ERRORLEVEL! == 0 (
+		echo [WARNING]: !PERL! not found, !classic_cpp_folder!\qos_string.h will not be updated.
+	) else (
+		perl !cStringifFile_script! !qos_file! PERFTEST_QOS_STRING > !classic_cpp_folder!\qos_string.h
```
Comment (javmorcas) [inline build.bat]: !PERL!

File: build.bat [hunk 3]
```diff
@@ -153,6 +156,28 @@ if !BUILD_JAVA! == 1 (
...
+		echo [INFO]: QoS String !classic_cpp_folder!\qos_string.h updated successfully.
+	)
+)
+
+if !BUILD_CPP03! == 1 (
+	call !PERL! -version > nul 2>nul
+	if not !ERRORLEVEL! == 0 (
+		echo [WARNING]: !PERL! not found, !modern_cpp_folder!\qos_string.h will not be updated.
+	) else (
+		perl !cStringifFile_script! !qos_file! PERFTEST_QOS_STRING > !modern_cpp_folder!\qos_string.h
```
Comment (javmorcas) [inline build.bat]: !PERL!

File: build.bat [hunk 4]
```diff
@@ -20,6 +22,7 @@ set MSBUILD_EXE=msbuild
 set JAVAC_EXE=javac
 set JAVA_EXE=java
 set JAR_EXE=jar
+set PERL=perl
```
Comment (javmorcas) [inline build.bat]: Add --perl as a command line parameter

File: build.sh [hunk 5]
```diff
@@ -11,12 +11,16 @@ modern_cpp_folder="${script_location}/srcCpp03"
...
+cStringifyFile_script="${script_location}/resource/perl_script/cStringifyFile.pl"
+qos_file="${script_location}/perftest_qos_profiles.xml"
+
 
 # Default values:
 BUILD_CPP=1
 BUILD_CPP03=1
 BUILD_JAVA=1
 MAKE_EXE=make
+PERL=perl
```
Comment (javmorcas) [inline build.sh]: Add options to pass the custom perl executable.

File: build.sh [hunk 6]
```diff
@@ -202,11 +206,31 @@ function additional_defines_calculation()
...
 }
 
+function geneate_qos_string()
+{
+    # If PERL is in the path, generate the qos_string.h file.
+    if [ "${BUILD_CPP}" -eq "1" ]; then
+        if [ -z `which "${PERL}"` ]; then
+            echo -e "${YELLOW}[WARNING]:${NC} ${PERL} not found, ${classic_cpp_folder}/qos_string.h will not be updated."
+        else
+            perl ${cStringifyFile_script} ${qos_file} PERFTEST_QOS_STRING > ${classic_cpp_folder}/qos_string.h
```
Comment (javmorcas) [inline build.sh]: use $PERL_EXE

File: build.sh (hunk 5 above)
Comment (javmorcas) [inline build.sh]: PERL_EXE

File: srcDoc/release_notes.rst [hunk 7]
```diff
@@ -80,6 +80,13 @@ Added RTI Perftest and RTI Connext DDS information at beginning of the test (#54
 Starting with this release, RTI Perftest will print at the beginning of the test
 its version and the version of RTI Connext DDS used to compile against.
 
+Update `qos_string.h` with the value of `perftest_qos_profiles.xml` in the build step(#63)
```
Comment (javmorcas) [inline srcDoc/release_notes.rst]: Add space

Comment (javmorcas) [reply srcDoc/release_notes.rst]: Automatically regenerate `qos_string.h` files if `perftest_qos_profiles.xml` is modified

File: srcDoc/release_notes.rst [hunk 8]
```diff
@@ -80,6 +80,13 @@ Added RTI Perftest and RTI Connext DDS information at beginning of the test (#54
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Starting with this release, RTI Perftest will udpate the file `qos_string.h`
```
Comment (javmorcas) [inline srcDoc/release_notes.rst]: the `qos_string.h` file

File: srcDoc/release_notes.rst [hunk 9]
```diff
@@ -80,6 +80,13 @@ Added RTI Perftest and RTI Connext DDS information at beginning of the test (#54
...
+with the value of `perftest_qos_profiles.xml` every time that *RTI Perftest* is
```
Comment (javmorcas) [inline srcDoc/release_notes.rst]: with the content of

File: resource/perl_script/cStringifyFile.pl [hunk 10]
```diff
@@ -0,0 +1,169 @@
+#########################################################################
```
Comment (javmorcas) [inline resource/perl_script/cStringifyFile.pl]: - Modify perl_script to scripts
- File a github issue to move java_scripts to scripts/java_execution_scripts

File: build.bat (hunk 1 above)
Comment (softwarejimenez) [reply build.bat]: Done

File: build.bat (hunk 2 above)
Comment (softwarejimenez) [reply build.bat]: Done

File: build.bat (hunk 3 above)
Comment (softwarejimenez) [reply build.bat]: Done

File: build.sh (hunk 5 above)
Comment (softwarejimenez) [reply build.sh]: Done

File: build.sh (hunk 6 above)
Comment (softwarejimenez) [reply build.sh]: Done

File: resource/perl_script/cStringifyFile.pl (hunk 10 above)
Comment (softwarejimenez) [reply resource/perl_script/cStringifyFile.pl]: Done

File: srcDoc/release_notes.rst (hunk 7 above)
Comment (softwarejimenez) [reply srcDoc/release_notes.rst]: Done

File: srcDoc/release_notes.rst (hunk 8 above)
Comment (softwarejimenez) [reply srcDoc/release_notes.rst]: Done

File: srcDoc/release_notes.rst (hunk 9 above)
Comment (softwarejimenez) [reply srcDoc/release_notes.rst]: done

File: build.bat (hunk 4 above)
Comment (softwarejimenez) [reply build.bat]: Done

**Comment by softwarejimenez** [comment]
@javmorcas can you take a look at it? Once you approved I will go ahead and merge it.

File: srcDoc/release_notes.rst [hunk 11]
```diff
@@ -80,6 +80,13 @@ Added RTI Perftest and RTI Connext DDS information at beginning of the test (#54
 Starting with this release, RTI Perftest will print at the beginning of the test
 its version and the version of RTI Connext DDS used to compile against.
 
+Automatically regenerate `qos_string.h` file if `perftest_qos_profiles.xml` is modified (#63)
```
Comment (javmorcas) [inline srcDoc/release_notes.rst]: Add to the compilation documentation this new parameter.

Comment (softwarejimenez) [reply srcDoc/release_notes.rst]: Done. @javmorcas  Can you review latest changes?

**Pull request merged**